pkg/cloud/services/networking/network.go

```diff
@@ -135,18 +136,17 @@ func (s *Service) ReconcileNetwork(openStackCluster *infrav1.OpenStackCluster, c
 		return nil
 	}
 
-	var opts createOpts
+	opts := createOpts{
+		AdminStateUp: gophercloud.Enabled,
+		Name:         networkName,
+	}
+
 	if openStackCluster.Spec.DisablePortSecurity {
-		opts = createOpts{
-			AdminStateUp:        gophercloud.Enabled,
-			Name:                networkName,
-			PortSecurityEnabled: gophercloud.Disabled,
-		}
-	} else {
-		opts = createOpts{
-			AdminStateUp: gophercloud.Enabled,
-			Name:         networkName,
-		}
+		opts.PortSecurityEnabled = gophercloud.Disabled
+	}
+
+	if openStackCluster.Spec.NetworkMTU > 0 {
+		opts.MTU = &openStackCluster.Spec.NetworkMTU
```

Comment:
Do we actually have to check if the OpenStack installation supports the `net-mtu` extension? Or can we assume a more or less up to date neutron version that supports setting the MTU?

Reply:
So you want to check for this extension by doing a equivalent of `openstack extension show net-mtu`.
If it exits, add this field, otherwise return an error to ask the user remove this field as it's not supported by the underlying provider ?

Reply:
Either way, this feels like a non-transient error.

Reply:
Yes, you're right, adding `requires net-mtu extension` in the comment should be sufficient!

Reply:
Field description is now updated :)